go/controller/autoscaler/autoscaler.go

```diff
@@ -182,6 +182,10 @@ func (a *Autoscaler) Monitor() {
 		select {
 		case <-a.ticker.C:
 		}
+		// TODO(helin): if we handle job add / delete by
```

Comment:
If `mutex.Lock` and `mutex.Unlock` spread in the code, it's possible to cause deadlock if not possible. In this case maybe is more suitable to use channel.